examples/index.md

```diff
@@ -162,3 +162,249 @@ Content-Type: application/vnd.api+json
   }
 }
 ```
+
+## Error Objects
+
+Examples of how [error objects](http://jsonapi.org/format/#error-objects) work.
+
+### Responses to failed resource creation:
+
+Response with one error object, with attribute error:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Invalid Attribute",
+      "source": { "pointer": "data/attributes/first-name" },
+      "detail": "First name must contain at least three characters."
+    }
+  ]
+}
+```
+
+Notice that the attribute `first-name` is dasherized.  Dasherized attributes are preferred
+over underscored.
+
+Notice also that the "detail" is specific to this occurrence of the problem,
+whereas the "title" is more generic.
+
+Notice that the "source" points to the associated attribute with the error.
+
+Response with one error object, with relationship error:
+
+```http
+HTTP/1.1 422 Unprocessable Entity
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Missing Association",
+      "source": { "pointer": "data/relationships/author" },
+      "detail": "Post author cannot be blank."
+    }
+  ]
+}
+```
+
+Notice that the "source" points to the associated relationship with the error.
+
+Response with one error object, with parameter error:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/vnd.api+json
+
+{
+  "errors": [
+    {
+      "title":  "Missing Query Parameter",
+      "source": { "parameter": "data" },
+      "detail": "The required parameter, data, is missing."
```

Comment:
> GET http://www.example.com/api/users/1/secrets.json could return a missing auth_key parameter?

That would be fine. I think returning an error (rather than ignoring) an invalid `fields` or `include` parameter would be fine too, but perhaps your example is safer.